src/Assets/Pack/Microsoft.AspNetCore.Assets.Pack.csproj

```diff
@@ -0,0 +1,212 @@
+
+    <!-- The package doesn't produce any lib or ref assemblies -->
+    <NoWarn>$(NoWarn);NU5128</NoWarn>
+
+    <StaticWebAssetBasePath>_framework</StaticWebAssetBasePath>
+    <DisableStaticWebAssetsBuildPropsFileGeneration>true</DisableStaticWebAssetsBuildPropsFileGeneration>
+    <DisableStaticWebAssetEndpointsBuildPropsFileGeneration>true</DisableStaticWebAssetEndpointsBuildPropsFileGeneration>
+    <StaticWebAssetsDisableProjectBuildPropsFileGeneration>true</StaticWebAssetsDisableProjectBuildPropsFileGeneration>
+    <StaticWebAssetsDisableProjectBuildMultiTargetingPropsFileGeneration>true</StaticWebAssetsDisableProjectBuildMultiTargetingPropsFileGeneration>
+    <StaticWebAssetsDisableProjectBuildTransitivePropsFileGeneration>true</StaticWebAssetsDisableProjectBuildTransitivePropsFileGeneration>
+    <StaticWebAssetsGetBuildAssetsTargets>_GetFrameworkBuildAssets</StaticWebAssetsGetBuildAssetsTargets>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <None Include="build\*" Pack="true" PackagePath="%(Identity)" />
+    <None Include="buildMultiTargeting\*" Pack="true" PackagePath="%(Identity)" />
+    <None Include="buildTransitive\*" Pack="true" PackagePath="%(Identity)" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <_BlazorJSContentRoot Condition="'$(Configuration)' == 'Debug'">..\..\Components\Web.JS\dist\Debug</_BlazorJSContentRoot>
+    <_BlazorJSContentRoot Condition="'$(Configuration)' == 'Release'">..\..\Components\Web.JS\dist\Release</_BlazorJSContentRoot>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <_BlazorJSFilename Include="blazor.web.js" />
+    <_BlazorJSFilename Include="blazor.server.js" />
+    <_BlazorJSFilename Include="blazor.webassembly.js" />
+
+    <_BlazorJSFile Include="@(_BlazorJSFilename->'$(_BlazorJSContentRoot)\%(Identity)')" />
+    <_BlazorJSMapFile Include="@(_BlazorJSFilename->'$(_BlazorJSContentRoot)\%(Identity).map')" />
+  </ItemGroup>
+
+  <Target Name="_CheckBlazorJSPath" AfterTargets="ResolveProjectReferences">
+
+    <ItemGroup>
+      <_MissingBlazorJSFile Include="@(_BlazorJSFile)" Condition="!EXISTS('%(_BlazorJSFile.FullPath)')" />
+      <_MissingBlazorJSMapFile Include="@(_BlazorJSMapFile)" Condition="!EXISTS('%(_BlazorJSMapFile.FullPath)')" />
+
+      <_AllMissingBlazorJSFiles Include="@(_MissingBlazorJSFile)" />
+      <_AllMissingBlazorJSFiles Include="@(_MissingBlazorJSMapFile)" />
+    </ItemGroup>
+
+    <Warning
+      Condition="'@(_AllMissingBlazorJSFiles)' != ''"
+      Text="'%(_AllMissingBlazorJSFiles.Identity)' does not exist. Falling back to checked-in copy." />
+
+    <ItemGroup>
+      <_BlazorJSFile Remove="@(_MissingBlazorJSFile)" />
+      <_BlazorJSFile Include="@(_MissingBlazorJSFile->'..\..\Components\Web.JS\dist\Release\%(Filename)%(Extension)')" />
+
+      <_BlazorJSMapFile Remove="@(_MissingBlazorJSMapFile)" />
+      <_BlazorJSMapFile Include="@(_MissingBlazorJSMapFile->'..\..\Components\Web.JS\dist\Release\%(Filename)%(Extension)')" />
+    </ItemGroup>
+
+  </Target>
```

Comment:
If we are no longer embedding the files, do we need this check?

Reply:
If we want to for now avoid removing the checked-in versions, that's fine, we can tackle that in a separate step, but I would expect nothing on the build to break because we removed this check or the checked-in files.

Ideally during source build this project would not build/do anything

Reply:
Yeah, I'll try removing the files from the repo. I'll still include the check to validate that the exist on disk, just so that there's a clear error if they haven't been built yet, but I'll remove the fallback logic.

Reply:
You'll have to avoid building this project during source build